packages/eas-cli/src/build/android/applicationId.ts

```diff
@@ -0,0 +1,158 @@
+      log.newLine();
+      log.warn(
+        `We detected that your Android project is configured with a different application id than the one defined in ${configDescription}.`
+      );
+      if (!(await hasApplicationIdInStaticConfigAsync(projectDir, exp))) {
+        log(`If you choose the one defined in ${configDescription} we'll automatically configure your Android project with it.
+However, if you choose the one defined in the Android project you'll have to update ${configDescription} on your own.
+Otherwise, you'll see this prompt again in the future.`);
+      }
+      log.newLine();
+      const { applicationIdSource } = await promptAsync({
+        type: 'select',
+        name: 'applicationIdSource',
+        message: 'Which application id should we use?',
+        choices: [
+          {
+            title: `${chalk.bold(
+              applicationIdFromAndroidProject
+            )} - In build.gradle in Android project`,
+            value: ApplicationIdSource.AndroidProject,
+          },
+          {
+            title: `${chalk.bold(applicationIdFromConfig)} - In your ${configDescription}`,
+            value: ApplicationIdSource.AppJson,
+          },
+        ],
+      });
+      switch (applicationIdSource) {
+        case ApplicationIdSource.AppJson: {
+          await updateApplicationIdInBuildGradleAsync(projectDir, exp);
+          break;
+        }
+        case ApplicationIdSource.AndroidProject: {
+          await updateAppJsonConfigAsync(projectDir, exp, applicationIdFromAndroidProject);
+          break;
+        }
+      }
+    }
+  } else if (!applicationIdFromAndroidProject && !applicationIdFromConfig) {
+    throw new Error(
+      `Please define "android.package" in ${configDescription} and run "eas build:configure" again.`
+    );
+  } else if (applicationIdFromAndroidProject && !applicationIdFromConfig) {
+    if (getConfigFilePaths(projectDir).staticConfigPath) {
+      await updateAppJsonConfigAsync(projectDir, exp, applicationIdFromAndroidProject);
+    } else {
+      throw new Error(
+        `Please define "android.package" in ${configDescription} and run "eas build:configure" again.`
+      );
+    }
+  } else if (!applicationIdFromAndroidProject && applicationIdFromConfig) {
+    // This should never happen, adding warning just in case
+    log.warn(
+      'applicationId is not specified in your ./android/app/build.gradle file. Make sure your project is confgured correctly before building.'
+    );
+    return;
+  }
+  if (allowExperimental) {
+    // package name does not need to be updated
+    await updatePackageNameAsync(projectDir, exp);
+    await gitAddAsync(path.join(projectDir, 'android'), { intentToAdd: true });
+  }
+}
+
+async function updateAppJsonConfigAsync(
+  projectDir: string,
+  exp: ExpoConfig,
+  newApplicationId: string
+): Promise<void> {
+  const paths = getConfigFilePaths(projectDir);
+  assert(paths.staticConfigPath, "can't update dynamic configs");
```

Comment:
we should provide guidance on what the user needs to update specifically, for example https://github.com/expo/expo-cli/blob/9777c36cbe73af2f4e4f47f408f083ac96c6be8b/packages/expo-cli/src/commands/upgrade.ts#L220-L228

eg: `Dynamic configuration files can't be updated automatically. Please set android.package to "${newApplicationId}`" and then (suggest next action here, eg: run build)`

Reply:
it's should never happen

I did not add any comments that state that because use assert should already inform about it.

Reply:
So maybe we should have this check/assert outside of this function? It's weird it's named `updateAppJsonConfigAsync` and there is an assert (checking if it's not dynamic config after all) inside.

Reply:
why?
name `updateAppJsonConfigAsync` already states that we want to update a static config (`AppJson`)